core/src/Revolution/modX.php

```diff
@@ -1751,18 +1751,21 @@ public function runProcessor($action = '', $scriptProperties = [], $options = []
             return $processor->run();
         }
 
-        // Check if we're dealing with an action like "context/update", which we transform into the 3.0+ class name
-        $legacyAction = 'MODX\\Revolution\\Processors\\' . implode('\\', array_map('ucfirst', explode('/', $action)));
+        // Check for a custom processors path that is not the core processor path to avoid running autoloaded core processors
+        if (!(isset($options['processors_path']) && !empty($options['processors_path']) && $options['processors_path'] != $this->config['processors_path'])) {
```

Comment:
This nested double negative condition had me squinting for a minute - any chance that can be simplified for future readers?

Reply:
A `or` condition (which is mostly the same as a `not and` condition) ist not possible without a PHP notice, since PHP stops the check in a `and` chain when the first check is false. The first check avoids the second check, throwing a notice. array_key_exist would be an option, but that’s slower.